### DISCUSSION TITLE
Different boundary shapes with differing potentials

### DISCUSSION THREAD

**Discussion by bokkaebi** [BLAST-WarpX/warpx#5287] [Simulation Inputs] [answered]
Is it possible to specify the shape of a boundary with a perhaps a function? For example, a sinusoidal boundary that removes particles. Also, is it possible to specify the potential at any given point of that boundary with a function? Thanks

**Comment by ax3l** [comment, marked as answer]
Hi @bokkaebi ,

Yes, you can do everything you describe: https://warpx.readthedocs.io/en/latest/usage/parameters.html#embedded-boundary-conditions

In `warpx.eb_implicit_function` you specify your sinusoidal embedded boundary profile and in `warpx.eb_potential(x,y,z,t)` your electric potential.

**Reply by KMalaravan** [reply]
Does  warpx.eb_potential(x,y,z,t) work with a cylindrical RZ grid or is there an equivalent?
